### PR TITLE
Expose all kolla commands to Enos

### PR DESCRIPTION
**Pull request by msimonin**
```
usage: enos kolla [-e ENV|--env=ENV] [-s|--silent|-vv] -- <command>...

Run arbitrary Kolla command.

Options:
  -e ENV --env=ENV     Path to the environment directory. You should
                       use this option when you want to link a specific
                       experiment.
  -h --help            Show this help message.
  -s --silent          Quiet mode.
  -vv                  Verbose mode.
  command              Kolla command (e.g prechecks, checks, pull)
```

Example :
```
enos kolla -- pull --tags nova
```

In addition to that, destroy now accepts a soft deploy mode (default mode) where
all containers, configuration and volumes are removed. It has also an optionnal
--include-images which will remove all the docker images.

In my opinion this is very very practical when it comes to iterate on kolla code
(or conf) :
```
enos deploy
while(not ok):
    <hack kolla code>
    enos destroy
    enos os
```

### CODE REVIEW

File: enos/enos.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
   ssh-tunnel     Print configuration for port forwarding with horizon.
   tc             Enforce network constraints
   info           Show information of the actual deployment.
-  destroy        Destroy the resources used by the deployment
+  destroy        Destroy the deployment and optionnaly the related ressurces.
```
Comment (denaitre) [inline enos/enos.py]: typos: `optionally` and `related resources`
Note that `ressource` is found multiple times in the rest of the commit

File: enos/enos.py [hunk 2]
```diff
@@ -583,20 +583,39 @@ def info(env=None, **kwargs):
...
+                    [--include-images]
+
+Destroy the deployment.
 
 Options:
   -e ENV --env=ENV     Path to the environment directory. You should
                        use this option when you want to link a specific
                        experiment [default: %s].
   -h --help            Show this help message.
+  --hard               Destroy the underlying ressources as well.
```
Comment (denaitre) [inline enos/enos.py]: `resources`

File: enos/enos.py [hunk 3]
```diff
@@ -583,20 +583,39 @@ def info(env=None, **kwargs):
...
   -s --silent          Quiet mode.
   -vv                  Verbose mode.
 """ % SYMLINK_NAME)
 @check_env
 def destroy(env=None, **kwargs):
-    provider = make_provider(env)
-    provider.destroy(CALL_PATH, env)
+    hard = kwargs['--hard']
+    if hard:
+        logging.info('Destroying all the ressources')
```
Comment (denaitre) [inline enos/enos.py]: `resources`

**Comment by msimonin** [comment]
@dpertin I've updated the patch.

**Pull request merged**